[FEATURE-SERVERSIDE-APPLY] Small integration fixes

apelisse opened the pull request:
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:
<!--  
If no, just write "NONE".
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

Commit:
FieldManager: Fix little bug/omission

This has been moved to a different method but was not removed.

Commit:
VersionConverter: Convert through internal version

One can't convert from external version to external version, so we
should go through internal version. Also make errors more explicit by
adding a little bit more info.

apelisse commented:
/assign @jennybuckley

apelisse commented:
/priority important-soon
/sig api-machinery

liggitt commented on staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/versionconverter.go:
```diff
@@ -56,10 +57,22 @@ func (v *versionConverter) Convert(object typed.TypedValue, version fieldpath.AP
+	// Convert to internal
+	internalVersion := schema.GroupVersion{
+		Group:   objectToConvert.GetObjectKind().GroupVersionKind().Group,
+		Version: runtime.APIVersionInternal,
```
What does this do for custom resource objects, which have no internal version?

liggitt replied:
https://github.com/kubernetes/kubernetes/pull/70087 plumbed a way for each handler to indicate the in-memory version it should deal with. Can we use that instead?

Commit:
FieldManager: use hub version everywhere

Commit:
fix toleration of FieldManager errors

We were updating the object to nil, so even though we tolerate errors,
we were creating errors further down the process.

BenTheElder commented:
/retest pull-kubernetes-godeps

Commit:
Remove ManagedFields after changing the version

But before we transform it into a typed object, so that it doesn't
appear in the typed object for proper operations, but also so that we
don't remove it from the original object that we receive as an argument.

BenTheElder commented:
/test pull-kubernetes-godeps

liggitt commented on staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:
```diff
@@ -367,8 +368,9 @@ func (p *smpPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (ru
+		if obj, err := p.fieldManager.Update(currentObject, newObj, "smPatcher"); err == nil {
+			newObj = obj
+		} else {
 			klog.Warningf("FieldManager: Failed to update object managed fields: %v", err)
```
surprised this isn't an error

apelisse replied:
Fixed.

jennybuckley replied:
We wanted to make sure that the feature couldn't break any other verbs besides apply so we are just logging the errors rather than returning them

liggitt replied:
I actually meant I was surprised it didn't return an error. Jenny's comment made sense, might be good to include that reasoning in a comment in the code in a future change (just logging an error there looks like a mistake)

liggitt commented:
plumbing of the hub version lgtm. I don't have the context to know if the current tests cover this change

Commit:
Make toleration warning into errors

apelisse commented:
> I don't have the context to know if the current tests cover this change

We could have better testing, that's next on the list :-)

liggitt commented:
/lgtm
